woocommerce/woocommerce-gateway-stripe
# Add new Stripe gateways UI

# Changes proposed in this Pull Request:

Resolves #1704 

This PR implements the new UI for the Stripe gateways outside of UPE. I made it reusable so that we can easily add all the gateways we have and others as they come in.

I've chose to build the new UI inside the UPE settings scope. Some reasons for that are: a) it can be considered part of the settings, even though it's not relying on UPE; b) the UI is very similar to what we have in UPE settings so we could reuse many of the components; c) I think it's not worth to create a new build/entrypoint to something that small, given its context as part of the settings and its reusability.

In the backend, there's a tradeoff I'm considering too. While taking advantage of the settings REST controller we already have, we'll be adding some new fields to the REST controller and that will bring more complexity and amplify its scope. However, the other option would be to create a new REST controller to handle the other gateways and I presume most (maybe all) of them will be removed soon, when UPE becomes mandatory, and this new controller will become useless. Besides, creating a new controller would force us to create a whole new structure in the React-side as well, which would become useless at some point either.

# Testing instructions
Ideally you must disable the new experience checkout so that you can test all the payment gateways, including those that come built-in with UPE.

- Ensure you have the option `_wcstripe_feature_upe_settings` set to `yes` in the database.
- Visit the [WooCommerce Payment settings page](http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout).
- Access any of the Stripe gateways settings, such as SEPA, Alipay, etc
- They should look similar to the screenshot below.
- You should be able to enable/disable and edit their name and description as you would like.
- Repeat the previous step and try to edit each payment gateway to confirm they're all working: SEPA, giropay, iDeal, Bancontact, EPS, SOFORT, P24, Alipay and Multibanco.

![Screen Shot 2021-08-19 at 9 00 33 AM](https://user-images.githubusercontent.com/273592/130082071-9424bbd7-3038-4ced-951f-2e22c9299268.png)

# What's still missing
- [x] Create issue to handle the webhook monitoring date

-------------------
